### PR TITLE
upgrade-threshold-crypto-libs

### PR DESCRIPTION
**Pull request by jkitman**
Pretty simple change to update the `threshold_crypto` related libs including `rand`.

I had to remove the mocks because they don't compile in the `threshold_crypto` branch, but it passes `cargo test --release`.

You need to use `--release` when testing or it'll take forever to do the real cryptography.

See also
https://github.com/fedimint/threshold_crypto/pull/3
https://github.com/fedimint/fedimint/pull/499

### CODE REVIEW

**Comment by jkitman** [comment]
Not sure why `clippy` and `fmt` are requiring so many changes now...

**Comment by jkitman** [comment]
@elsirion @dpc should we merge this into master then?  It nukes the `mock-crypto` testing crate.

Related to https://github.com/fedimint/fedimint/issues/1746

File: Cargo.toml [hunk 1]
```diff
@@ -31,23 +31,23 @@ env_logger = "0.9.0"
...
 init_with = "1.1.0"
 log = "0.4.17"
-rand = "0.7"
+rand = "0.8.5"
 rand_derive = "0.5.0"
 reed-solomon-erasure = "5.0.3"
 serde = { version = "1.0.137", features = ["derive", "rc"] }
 thiserror = "1.0.31"
-threshold_crypto = { git = "https://github.com/fedimint/threshold_crypto" }
+threshold_crypto = { git = "https://github.com/elsirion/threshold_crypto", branch = "2022-08-upgrade-pairing" }
```
Comment (elsirion) [inline Cargo.toml]: We probably want to change this back now after fedimint/threshold_crypto#6 got merged?

File: Cargo.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ rand_derive = "0.5.0"
 reed-solomon-erasure = "5.0.3"
 serde = { version = "1.0.137", features = ["derive", "rc"] }
 thiserror = "1.0.31"
-threshold_crypto = { git = "https://github.com/elsirion/threshold_crypto", branch = "2022-08-upgrade-pairing" }
+threshold_crypto = { git = "https://github.com/jkitman/threshold_crypto", branch = "upgrade-threshold-crypto-libs" }
```
Comment (elsirion) [inline Cargo.toml]: change to fedimint?

File: hbbft_testing/Cargo.toml [hunk 3]
```diff
@@ -28,4 +28,4 @@ proptest = "1.0.0"
 rand = "0.8.5"
 rand_xorshift = "0.3.0"
 thiserror = "1.0.31"
-threshold_crypto = { git = "https://github.com/elsirion/threshold_crypto", branch = "2022-08-upgrade-pairing" }
+threshold_crypto = { git = "https://github.com/jkitman/threshold_crypto", branch = "upgrade-threshold-crypto-libs" }
```
Comment (elsirion) [inline hbbft_testing/Cargo.toml]: change to fedimint?

**Comment by jkitman** [comment]
@elsirion Should be good to go now.

**Review by dpc — APPROVED** [review]

**Review by elsirion — APPROVED** [review]

**Pull request merged**